marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/ModelController.java

```diff
@@ -76,11 +84,85 @@
     @RequestMapping(value = "/{modelName}/entityTypes", method = RequestMethod.PUT)
     @ApiImplicitParam(required = true, paramType = "body", dataType = "ModelDefinitions")
     public ResponseEntity<Void> updateModelEntityTypes(@ApiParam(hidden = true) @RequestBody JsonNode entityTypes, @PathVariable String modelName) {
+        // update the model
         newService().updateModelEntityTypes(modelName, entityTypes);
+
+        //deploy updated configs
+        deployModelConfigs();
+
         return new ResponseEntity<>(HttpStatus.OK);
     }
 
 
+    void deployModelConfigs() {
+        ManageClient manageClient = hubClientProvider.getHubClient().getManageClient();
+        DatabaseClient finalDatabaseClient = hubClientProvider.getHubClient().getFinalClient();
+        DatabaseClient stagingDatabaseClient = hubClientProvider.getHubClient().getStagingClient();
+
+        JsonNode modelConfigNode = newService().generateModelConfig();
+
+        // Deploy search options
+        String explorerOptions = modelConfigNode.get("search-options").get("explorer").asText();
```

Comment:
Let's put each of these deploy steps in a separate private method and then put each call to ML in a try-catch iof the form:

```
try {
     make a single call to ML - e.g. to deploy explorer search options to staging
} catch (Exception ex) {
     throw new RuntimeException("Unable to deploy explorer search options to staging after updating entity models: cause: " + ex.getMessage(), ex);
}
```

The reason for this is that the typical errors we get back from the Manage API and REST API don't give a ton of context and often lead to the conclusion of "Gradle is broken, we must replace it". Gradle isn't being used here, so hopefully it won't be blamed, but providing some context around the source of the error will be helpful.

Reply:
Right, updated in here https://github.com/marklogic/marklogic-data-hub/pull/3941/commits/b6777afc22dfc3a5bd4d452ff1d0df426f91e0cc